CoreScriptsRoot/ServerCoreScripts/ServerChat/ChatServiceProxy.lua

```diff
@@ -0,0 +1,212 @@
+		end
+		
+		obj.Target:RemoveSpeaker(speakerName)
+	end
+	
+	function obj:GetSpeaker(speakerName)
+		AssertParameterTypeEquals(speakerName, "string")
+		local speaker = obj.Target:GetSpeaker(speakerName)
+		return speaker and SpeakerProxy.CreateProxy(speaker) or nil
+	end
+	
+	function obj:GetAutoJoinChannelList()
+		local list = obj.Target:GetAutoJoinChannelList()
+		local proxyList = {}
+		for i, channel in pairs(list) do
+			table.insert(proxyList, ChatChannelProxy.CreateProxy(channel))
+		end
+		return proxyList
+	end
+	
+	function obj:AddWordFilter(expression)
+		AssertParameterTypeEquals(expression, "string")
+		obj.Target:AddWordFilter(expression)
+	end
+	
+	function obj:RemoveWordFilter(expression)
+		AssertParameterTypeEquals(expression, "string")
+		obj.Target:RemoveWordFilter(expression)
+	end
+	
+	function obj:AddWordAlias(expression, replacement)
+		AssertParameterTypeEquals(expression, "string")
+		AssertParameterTypeEquals(replacement, "string")
+		obj.Target:AddWordAlias(expression, replacement)
+	end
+	
+	function obj:RemoveWordAlias(expression)
+		AssertParameterTypeEquals(expression, "string")
+		obj.Target:RemoveWordAlias(expression)
+	end
+	
+	function obj:GetChannelList()
+		return obj.Target:GetChannelList()
+	end
+	
+	function obj:RegisterFilterMessageFunction(funcId, func)
+		AssertParameterTypeEquals(funcId, "string")
+		AssertParameterTypeEquals(func, "function")
+		
+		if (funcId == "default_filter") then
+			error("You cannot register this filter!")
+		end
+		
+		local allow = false
+		pcall(function()
+			local ret = func("Player1", "test message")
+			AssertParameterTypeEquals(ret, "string")
+			allow = true
+		end)
+		
+		if (not allow) then
+			error("Function must return a string!")
+		end
+		
+		obj.Target:RegisterFilterMessageFunction(funcId, func)
+	end
+
+	function obj:UnregisterFilterMessageFunction(funcId, func)
+		AssertParameterTypeEquals(funcId, "string")
+		
+		if (funcId == "default_filter") then
+			error("You cannot unregister this filter!")
+		end
+		
+		obj.Target:UnregisterFilterMessageFunction(funcId)
+	end
+	
+	function obj:RegisterProcessCommandsFunction(funcId, func)
+		AssertParameterTypeEquals(funcId, "string")
+		AssertParameterTypeEquals(func, "function")
+		
+		if (funcId == "default_commands") then
+			error("You cannot register this set of commands!")
+		end
+		
+		obj.Target:RegisterProcessCommandsFunction(funcId, func)
+	end
+	
+	function obj:UnregisterProcessCommandsFunction(funcId)
+		AssertParameterTypeEquals(funcId, "string")
+		
+		if (funcId == "default_commands") then
+			error("You cannot unregister this set of commands!")
+		end
+		
+		obj.Target:UnregisterProcessCommandsFunction(funcId)
+	end
+	
+	local metatable = getmetatable(proxy)
+	metatable.__metatable = "The metatable is locked"
+	metatable.__tostring = function () return tostring(obj.Target) end
+	
+	metatable.__index = function(tbl, index)
+		if (index == "ChatLogLength" or
```

Comment:
This is an awful lot of `or`s flying around here. Can we put this in a lookup table?